### PR TITLE
Use debian9 on g5k and vagrant

### LINKED ISSUE #204: [stable/pike] no network connectivity between netns
> Note that it is recommended to use Debian 9 (stretch) as base image for your deployment as you may experience some issue with the network connectivity of your VMs.
It has been chosen to keep Debian 8 (jessie) for `3.0.0` but you can still upgrade your base image to use the latest Debian. Please make sure to change your network interface name mapping in the reservation file accordingly to your environment.

Historical issue below :
Let's try to narrow down this issue

# Observation

When deploying stable/pike on g5k, there is no network connectivity between the various netns enforced by neutron.
For example we can't ping the gateway (`192.168.0.1` in the router namespace) from the dhcp server namespace (`192.168.0.2`).
This result in VMs not getting their dhcp leases.

# Tests : 

## work with

* `enos 2.3.0` (stable/ocata), docker images `beyondtheclouds:4.0.3`, provider `g5k`
* `enos stable/pike on ubuntu16.04`
* `enos stable/pike on debian9`

## doesn't work with

* `enos stable/pike` and docker images `beyondtheclouds:5.0.1`, provider `g5k`
* `enos stable/pike` and docker images `beyondtheclouds:5.0.1`, provider `vagrant` (`debian/contrib-jessie64`, `ovs`) [comment 1]
* `enos stable/pike` and docker images `beyondtheclouds:5.0.1`, provider `vagrant` (`debian/contrib-jessie64`, `linuxbridge`) [comment 2]

### PR DESCRIPTION
**Pull request by rcherrueau**
Fix #204

### CODE REVIEW

File: enos/ansible/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -17,3 +17,15 @@
 
 - name: Mount /run
   command: mount --make-shared /run
+
+- name: Ensure nscd is not loaded
+  shell: systemctl list-units --all nscd.service |fgrep "0 loaded units listed"
```
Comment (msimonin) [inline enos/ansible/roles/common/tasks/main.yml]: Ansible service module cannot be used ?

File: enos/provider/enos_vagrant.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ def destroy(self, env):
     def default_config(self):
         return {
             'backend': 'virtualbox',
-            'box': 'debian/jessie64',
+            'box': 'bento/debian-9',
             'user': 'root',
-            'interfaces': ('eth1', 'eth2')
+            'interfaces': ('enp0s8', 'enp0s9')
```
Comment (msimonin) [inline enos/provider/enos_vagrant.py]: Hail enoslib :)

File: enos/templates/reservation.yaml.sample [hunk 3]
```diff
@@ -51,7 +51,9 @@ resources:
 # ############################################### #
 provider:
   type: vagrant
-  box: debian/contrib-jessie64
+  box: bento/debian-9
```
Comment (msimonin) [inline enos/templates/reservation.yaml.sample]: We would like to change also in tests/functionnal/packaging,vbox,g5k

File: enos/ansible/roles/common/tasks/main.yml (hunk 1 above)
Comment (rcherrueau) [reply enos/ansible/roles/common/tasks/main.yml]: It change nothing. Same problem, `service` and `systemd` module failed if the service is not loaded.

File: enos/templates/reservation.yaml.sample (hunk 3 above)
Comment (rcherrueau) [reply enos/templates/reservation.yaml.sample]: I do a `find -name 'reservation.yaml` and only found `./tests/functionnal/tests/packaging/reservation.yaml` that use static provider.

File: enos/ansible/roles/common/tasks/main.yml (hunk 1 above)
Comment (msimonin) [reply enos/ansible/roles/common/tasks/main.yml]: One idea could be to use the service module and ignore the error (if any).

https://docs.ansible.com/ansible/latest/playbooks_error_handling.html#ignoring-failed-commands

Comment (rcherrueau) [reply enos/ansible/roles/common/tasks/main.yml]: done

File: enos/templates/reservation.yaml.sample (hunk 3 above)
Comment (msimonin) [reply enos/templates/reservation.yaml.sample]: * https://github.com/BeyondTheClouds/enos/blob/master/tests/functionnal/tests/grid5000/grid5000.yaml
* https://github.com/BeyondTheClouds/enos/blob/master/tests/functionnal/tests/vbox/vbox.yaml

Comment (rcherrueau) [reply enos/templates/reservation.yaml.sample]: Done.

Thanks for the links:
- `grid5000.yaml` does not mention the `env_name`, so it should use the default one.
- I remove `box` from `vbox.yaml`, so it should use the default one.

**Pull request merged**